eoc_journal/eoc_journal.py

```diff
@@ -97,6 +106,49 @@ def student_view(self, context=None):
         )
         return fragment
 
+    def list_user_pb_answers_by_section(self):
+        """
+        Returns a list of dicts with pb-answers grouped by section.
+        """
+        # Get the selected blocks and their answers
+        blocks = self.list_pb_answers()
```

Comment:
@snowcrshd Sorry I did not notice this earlier, but we have to make sure to only select the questions that the staff user selected when editing the block in the Studio. This is untested but I guess it would looks like this:

```python
blocks = self.list_pb_answers()
blocks = [block for block in blocks if block['id'] in self.selected_pb_answer_blocks]`